src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqConnectionSettings.cs

```diff
@@ -0,0 +1,177 @@
+                UserName = UserName,
+                Password = Password,
+                Ssl = SslOption,
+                AutomaticRecoveryEnabled = AutomaticRecoveryEnabled
+            };
+        }
+
+        private static string GetQueueNameFromUri(Uri uri)
+        {
+            if (uri.Segments.Length < 2)
+            {
+                throw new ArgumentException($"Expected a non-empty segment in the uri {uri}");
+            }
+
+            var lastSegment = uri.Segments.LastOrDefault();
+            if (lastSegment.EndsWith("/"))
+            {
+                // Exchange found but no queueName, so create unique queueName
+                return $"{TempQueuePrefix}{new Guid()}";
+            }
+
+            return lastSegment.Replace("/", string.Empty);
+        }
+
+        private static string GetExchangeFromUri(Uri uri)
+        {
+            if (uri.Segments.Length < 2)
+            {
+                throw new ArgumentException($"Expected a non-empty segment in the uri {uri}");
+            }
+
+            var secondSegment = uri.Segments[1];
+            if (secondSegment.EndsWith("/"))
+            {
+                // Exchange name was found
+                return secondSegment.Replace("/", string.Empty);
+            }
+
+            // Exchange name was not found
+            return string.Empty;
+        }
+
+        private static string GetRoutingKeyFromUri(Uri uri, string queueName)
+        {
+            if (string.IsNullOrEmpty(uri.Fragment))
+            {
+                return queueName;
+            }
+
+            return uri.Fragment.Replace("#", string.Empty);
+        }
+
+        private static string SetUserName(Uri uri, ICredentials credentials)
+        {
+            var userNameFromCredentials = credentials?.GetCredential(uri, string.Empty).UserName;
+            if (!string.IsNullOrEmpty(userNameFromCredentials))
+            {
+                return userNameFromCredentials;
+            }
+
+            return GetUserNameFromUri(uri);
+        }
+
+        private static string GetUserNameFromUri(Uri uri)
+        {
+            if (string.IsNullOrEmpty(uri.UserInfo))
+            {
+                return ConnectionFactory.DefaultUser;
+            }
+
+            return uri.UserInfo.Split(':').First();
+        }
+
+        private static string SetPassword(Uri uri, ICredentials credentials, SslOption sslOption)
+        {
+            if (!string.IsNullOrEmpty(sslOption.CertPath) || sslOption.Certs?.Count > 0)
+            {
+                return string.Empty;
+            }
+
+            var passwordFromCredentials = credentials?.GetCredential(uri, string.Empty).Password;
+            if (!string.IsNullOrEmpty(passwordFromCredentials))
+            {
+                return passwordFromCredentials;
+            }
+
+            return GetPasswordFromUri(uri);
+        }
+
+        private static string GetPasswordFromUri(Uri uri)
+        {
```

Comment:
I don't believe we should support or encourage encoding the password as part of the uri. It's a bad idea from a security perspective. Uri's can get logged in log files for example.